tests/base_real_sensor_mock_test_case.py

```diff
@@ -116,7 +118,7 @@ def read(self, length: int) -> str:
         else:
             return ch.encode(self.sensor.charset)
 
-    async def read_next(self, timeout: float = STD_TIMEOUT) -> str:
+    async def read_next(self, timeout: float = STD_TIMEOUT) -> None:
```

Comment:
With this name I would certainly expect it to return data (as the old version _looks like_ it was meant to do -- I don't know if it actually did).

If you just want it to wait for an event then please consider a different name.

Also please explain what this method does in the doc string. "Helper method to ease..." is more of an explanation of why it exists than what it does. Such information might be useful as a detail, but not so much as the main doc.